src/tests/account/ethereum/test_dual_eth_account.cairo

```diff
@@ -167,114 +172,68 @@ fn test_dual_supports_interface_exists_and_panics() {
 //
 
 #[test]
+#[ignore] // REASON: foundry entrypoint_not_found error message inconsistent with mainnet.
 fn test_dual_setPublicKey() {
     let (camel_dispatcher, target) = setup_camel();
-    let new_public_key = NEW_ETH_PUBKEY();
+    let contract_address = camel_dispatcher.contract_address;
+
+    start_cheat_caller_address(contract_address, contract_address);
 
-    testing::set_contract_address(camel_dispatcher.contract_address);
+    let key_pair = KEY_PAIR();
+    let signature = get_accept_ownership_signature(contract_address, ETH_PUBKEY(), key_pair);
 
-    camel_dispatcher.set_public_key(new_public_key, get_accept_ownership_signature_camel());
-    assert_eq!(target.getPublicKey(), new_public_key);
+    camel_dispatcher.set_public_key(key_pair.public_key, signature);
+    assert_eq!(target.getPublicKey(), key_pair.public_key);
 }
 
 #[test]
-#[should_panic(expected: ("Some error", 'ENTRYPOINT_FAILED',))]
+#[ignore] // REASON: foundry entrypoint_not_found error message inconsistent with mainnet.
```

Comment:
Not really, is the other way around. This test would work if the foundry message were consistent with mainnet, because we could update the `try_selector_with_fallback` function to match the actual error in line 28 of utils.cairo.

Reply:
I see, makes sense